### DISCUSSION TITLE
Habe einen Wunsch, jk-bms Umsetzung auf CAN Victron

### DISCUSSION THREAD

**Discussion by testpaul999** [shining-man/bsc_fw#24] [General]
Hallo,

im Wiki steht „Zum Testen kann aber auch ein ESP32-Dev-Kit (z.B. ein NodeMCU mit einem ESP-WROOM-32) genutzt werden“.
An welchem Gpio wird die Uart des jk-bms angeschlossen?
An welchem Gipo wird der CAN Wandler [TJA1050 CAN] angeschlossen?
Ich würde gerne nur eine Teil der Funktion nutzen und zwar Jk-Bms RS232 auf CAN  für Victron wandeln.
Warum? Ich möchte mir einen USB Port sparen und mit dem CAN-Head arbeiten.
Ist das machbar?
Danke schon mal fürs lesen.

Grüße

**Comment by shining-man** [comment]
Das ist so machbar und funktioniert prinzipiell auch. Ich empfehle es aber nicht, da du dann keine galvanische Trennung hast. Das ist der Vorteil und der Grund für die BSC Hardware.

Folgendes sind die GPIOs für CAN und Serial 0:
CAN:
TX: GPIO4
RX: GPIO5

Serial 0:
RX: GPIO16
TX: GPIO17

**Reply by testpaul999** [reply]
Danke und gut zu wissen. 👍🏻

**Comment by testpaul999** [comment]
Hallo, 
Funktioniert 🙂
Nur scheint die Batterietemperatur auf der Übersichtsseite nicht zu passen.
![image](https://user-images.githubusercontent.com/35597907/217650983-dd61b297-e3d4-4471-b9fd-aaddee1851b0.jpg)
Im Jk-bms sieht es so aus.
![B465B8C6-37E3-4706-8A9B-F13566661780](https://user-images.githubusercontent.com/35597907/217651133-a816afd0-0bc5-4da2-abc2-30cbc685edc1.png)
Und die Zellen Spannungen passen in der Übersicht nicht.
![image](https://user-images.githubusercontent.com/35597907/217651691-d5ed552d-45b8-47cf-89d9-3a9f0a9aa505.jpg)

grüße

**Reply by shining-man** [reply]
Für die Batterietemperatur wird die erste Temperatur genommen die das BMS liefert. Dies ist beim JK die MOS Temperatur. 
Evtl. werde ich dies irgendwann noch einstellbar machen, dass man hier auch einen angeschlossenen Onewire Sensor nehmen kann.

Was passt bei der Zellenübersicht nicht? Stimmen die Min/Max-Werte nicht?
Die Spannungen der Einzelzellen ist aktuell nicht implementiert, da ich nicht die CAN-Adresse weiß. Ich werde in den nächsten Monaten ein VenusOS-Testsystem aufbauen, da werde ich dies dann einmal probieren.
Wenn du die CAN-Adresse hast, kann ich es problemlos einbauen.

**Reply by testpaul999** [reply]
Hallo,
Danke für das Feedback.
Das mit der Temperatur ist „blöd“ ich würde gerne, da ich nur einen Akku habe, den esp32 nur als Übersetzer nehmen.
Die CAN Adressen habe ich leider nicht.

dann warte ich mal gespannt auf die nächsten Neuerungen.
Grüße

**Reply by shining-man** [reply]
Welche Temperatur möchtest du denn angezeigt haben?

**Reply by testpaul999** [reply]
T1 oder T2 wenn Du so fragst, wählbar im UI. 🤷🏻‍♂️

**Reply by shining-man** [reply]
Kommt in nächste Release mit rein

**Reply by testpaul999** [reply]
Danke 👍🏻
Wenn ich irgendwie helfen kann…. Gerne

**Comment by shining-man** [comment]
Ich hab die Erweiterung eingebaut und das ganze gepusht. Du kannst es testen.
Anbei noch das bin File für das Update.
[firmware.zip](https://github.com/shining-man/bsc_fw/files/10712693/firmware.zip)

**Reply by testpaul999** [reply]
Guten Morgen, ich habe die Firmware mal eingespielt und mir angesehen.
Das mit der Sensor Auswahl für die Temperaturüberwachung finde ich gut.
Aber ich denke du hast ein kleinen Beschriftungsfehler in den seriellen Schnittstellen drin. Unter der Reiterkarte Schnittstellen heißt es Seriell 0-2.
Bei den Einstellungen für den Wechselrichter heißt es Serie 1-3. Ich denke das ist nur ein Typo.
Ach so, und bei den Alarm steht nur noch Bluetooth, aber nicht so seriell in der Überschrift.
Da aber darunter eine Schaltfläche für die Auswahl der Schnittstelle ist, würde ich wieder Bluetooth noch seriell dort oben hinschreiben, sondern nur Alarmregeln
Das hier:
> Bluetooth Alarmregeln

Alarmregel 0
--
Wäre es möglich, die blaue LED im Takt, der ein gelesenen oder ausgegebenen Daten flackern zu lassen so leuchtet nur die Betriebs LED und man weiß nicht so ganz genau ob der ESP32 arbeitet.


grüße

**Reply by shining-man** [reply]
Aktuell blink die blaue LED immer dann wenn der Task mit den Alarmregeln bearbeitet wird. Damit soll erkennbar sein, dass der Task für die Alarmregeln noch läuft und nicht hängt. 
Da du aber nicht die orginale BSC Hardware nutzt, sondern ein Dev. Board ist die LED natürlich an einem anderen GPIO. Wenn du anstatt Serial 0 Serial 1 nutzt, dann sollte die LED blinken.

**Comment by testpaul999** [comment]
Danke
Gpio 9 und 10?
Also rx0 und tx0?

**Reply by shining-man** [reply]
Serial 1:
RX: GPIO23
TX: GPIO25

**Comment by testpaul999** [comment]
Echt 23 und 25?
Nutze diesen hier 
https://circuitstate.com/pinouts/doit-esp32-devkit-v1-wifi-development-board-pinout-diagram-and-reference/#PNG

Danke

**Reply by shining-man** [reply]
Was du benutzt ist eigentlich egal.
Entscheidend ist was ich in der Software festgelegt habe.

//Serial 0
#define SERIAL1_PIN_RX               16
#define SERIAL1_PIN_TX               17
#define SERIAL1_PIN_TX_EN            18

//Serial 1
#define SERIAL2_PIN_RX               23
#define SERIAL2_PIN_TX               25
#define SERIAL2_PIN_TX_EN             2

//Serial 2
#define SERIAL3_PIN_RX               35
#define SERIAL3_PIN_TX               33
#define SERIAL3_PIN_TX_EN            32

**Reply by testpaul999** [reply]
👍🏻💡

**Comment by testpaul999** [comment]
Hallo,

die strommessung ist um den Faktor 10 falsch.
Habe mit 1.8A getestet und angezeigt wurden 180A.

Grüße

**Reply by shining-man** [reply]
Hier wären schon ein paar mehr Infos wichtig, denn so kann ich das nicht nachvollziehen.
Welches BMS verwendest du?
Wo wird der 10 fache Strom angezeigt? (180A wäre das 100fache von 1,8A)
Kommt der falsche Strom auch per MQTT, oder nutzt du das gar nicht?
Wenn ein Problem besteht, dann bitte mit möglichst vielen Details, da ich sonst auch nicht helfen kann. Ich habe nicht alle Hardware selber vorrätig um das ganze zu testen.

**Reply by testpaul999** [reply]
Hallo,
Sorry habe mich vertippt. 100 fach ist richtig.
Also 
JK-BMS 
JK-B2A20S20P 
FW Version 11.x
Anbindung seriell an z.B. 
Serial 0:
RX: GPIO16
TX: GPIO17
aber auch per Serial 1:
RX: GPIO23
TX: GPIO25
Dann CAN zum RPI mit Venus-OS (Victron) auf CAN-Head.
Damit erschient das BMS quasi automatisch im Victronsystem.
Und genau hier wird auch der 100 fache Wert des Stroms angezeigt!
Die Gegenkontrolle erfolgte über die APP von JK, welche 1.8A angezeigt hat.
Nein MQTT nutze ich derzeit nicht.
Wenn ich irgendwie debuggen kann oder soll sage Bescheid.

Und wie unten beschrieben ist die CAN Verbindung nicht besonders stabil.
Ja der CAN Head hat einen Abschlusswiederstand und das Kabel ist geschirmt.
Auf der selben Hardware läuft die unten genannte Implementierung stabil.
Nur leider ist die nicht so toll in der Integration und Einstellung wie die von Dir.

Grüße aus Berlin

**Reply by shining-man** [reply]
Ich macheinmal eine Version mit aktiven Debugausgaben. Dann sehe ich die Werte die er vom BMS bekommt und die er an den Wechslrichter (Venus) sendet. Gibt es hier Einstellungen im VenusOS, die den Wert beeindflussen können?

**Reply by shining-man** [reply]
[firmware.zip](https://github.com/shining-man/bsc_fw/files/10736644/firmware.zip)
Den Log der seriellen-Ausgabe dann in einer Datei zurück.

**Comment by testpaul999** [comment]
Guten Tag, ich habe da mal eine Beobachtung gemacht.
Mit den hier verwendeten gpio Port für can und dem tja1050 habe ich eine sehr instabile Verbindung.
Wenn ich eine andere Umsetzung nehme 
https://github.com/Uksa007/esphome-jk-bms-can/discussions/1#discussioncomment-4872226
also unten benannte Ports ist die CAN Verbindung stabil. 
# ESP32 Serial port pins:
# GPIO pins your CAN bus transceiver(TJA1050) is connected to the ESP, note! TX->TX and RX->RX. 
  can_tx_pin: GPIO23
  can_rx_pin: GPIO22

liesse sich bitte bitte eine Version linken, die diese Ports nutzt?
Danke

**Reply by shining-man** [reply]
Was heist die Verbindung ist instabil? Vor allem was ist hierbei instabil. Die Verbinung zum BMS oder zum Wechselrichter oder...?
Wie zeigt sich die instabile Verbindung?
Wenn du Hilfe brauchst, dann brauche ich alle Details.

Wenn es Probleme bei der Hardware gibt, helfe ich prinzipiell nur dann, wenn die BSC Hardware genutzt wird. Bei diesen ganzen billigen China-ESP-Boards habe ich schon so viel wilde Dinge erlebt. 
Wenn du andere GPIOs willst, dann kannst du in der Canbus.cpp in Zeile 114 [CAN.setPins(5,4);] die GPIOs ändern, obwohl ich darin keinen wirklichen SInn sehe, denn die Matrix im ESP32 schaltet die Funktion auf den GPIO wo es ihr gesagt wird.

**Reply by testpaul999** [reply]
![image](https://user-images.githubusercontent.com/35597907/218858711-f52772c1-5689-49ef-84ad-6109014f7516.jpg)
Hallo,
Danke für die FW werde testen.
Instabil ist die CAN Verbindung, nehme ich an.
Zumindest verschwindet das BMS immer wider aus den Venus-OS, um dann wider zu erscheinen usw..
Werden CAN Daten nur gesendet, wenn auch Serielle Daten kommen, ich meine bleibt der CANBus auch aktiv wenn keine Daten vom BMS kommen? Oder wird er dann abgeschaltet?
Weil dann verschwindet auch das BMS aus den Victron System.

**Comment by testpaul999** [comment]
Hallo,

Ich bin im coden echt eine Niete, daher verzeih die Frage, mit welchem Programm kann ich das ganze Kompilieren?
„Wenn du andere GPIOs willst, dann kannst du in der Canbus.cpp in Zeile 114 [CAN.setPins(5,4);] die GPIOs ändern“

Leider funktioniert das Log-file nicht.
![image](https://user-images.githubusercontent.com/35597907/218863869-3dbcdac7-3920-4de9-b191-9fbba8c40c1e.jpg)
BMS ist da
![F34C39E5-C476-4987-820D-CC7A970F4CEF](https://user-images.githubusercontent.com/35597907/218864054-e001a4b7-30fd-4378-9ed6-216f0afad106.jpeg)
Kurz darauf ist es weg.
![684F5C6D-3968-4692-B84B-20C9213D597F](https://user-images.githubusercontent.com/35597907/218864130-2d9cc655-0cd3-4435-940c-34f538c27a2d.jpeg)
Und das toggles immer hin und her
Hier die stromanzeige
![759A64D2-B294-4423-99F1-9807E6930C9C](https://user-images.githubusercontent.com/35597907/218864358-81623e07-1d51-4bc1-a92a-bb922334595b.jpeg)

und das ist das bms was ich nutze

![CC2E9151-2B65-4FEF-9314-804C43A047C5](https://user-images.githubusercontent.com/35597907/218864525-1d1e9ea6-b761-4a04-b114-eeb73b78218e.jpeg)

**Reply by testpaul999** [reply]
Ps. Bis morgen…. Gute Nacht

**Reply by shining-man** [reply]
> Leider funktioniert das Log-file nicht.

Wie ich geschrieben hatte kommt das Log in dem Fall über die serielle Schnittstelle.

**Comment by testpaul999** [comment]
Ok, über welche?
Serial 0 nehme ich ja zu Kommunikation mit dem bms.

Viele Grüße

Jens-Uwe Franz

Am 15.02.2023 um 05:39 schrieb shining-man ***@***.***>:

﻿

Leider funktioniert das Log-file nicht.

Wie ich geschrieben hatte kommt das Log in dem Fall über die serielle Schnittstelle.

**Reply by shining-man** [reply]
Den ganz normalen USB-Port von deinem ESP32 Board.

**Comment by testpaul999** [comment]
👍🏻

Viele Grüße

Jens-Uwe Franz

Am 15.02.2023 um 07:01 schrieb shining-man ***@***.***>:

﻿

Den ganz normalen USB-Port von deinem ESP32 Board.

**Comment by testpaul999** [comment]
Kommt heute Abend, muss heute ins Office ..,

Grüße
Jens - Uwe Franz

Am 15.02.2023 um 07:01 schrieb shining-man ***@***.***>:

﻿

Den ganz normalen USB-Port von deinem ESP32 Board.

**Reply by testpaul999** [reply]
Guten Abend,

hier das log file 
[bsc.log](https://github.com/shining-man/bsc_fw/files/10746611/bsc.log)
Und der Screenshot der BSM App 
![23E43C03-DB0C-4BC7-B255-A3221CD2BBC3](https://user-images.githubusercontent.com/35597907/219131585-1af78204-41f4-4c52-b45f-318d28778ae6.jpeg)
Grüße

**Reply by shining-man** [reply]
Ich hab die Logausgabe noch etwas erweitert. Kannst du mir noch einmal einen Log schicken.

[firmware.zip](https://github.com/shining-man/bsc_fw/files/10746749/firmware.zip)

**Reply by testpaul999** [reply]
[bsc_log2.log](https://github.com/shining-man/bsc_fw/files/10746966/bsc_log2.log)
Da wo die 180 erscheinen ist es entladen.
![B5D80248-73D7-4511-8DF6-60B778F6DB82](https://user-images.githubusercontent.com/35597907/219139609-2e101135-b124-454b-b6e6-5ecb8642642e.png)

Am Schluss gibt es dann 0x84:128, 30, 32798.00000 da lade ich.

![32822499-0847-42C9-8FCF-0D0EDBA630F8](https://user-images.githubusercontent.com/35597907/219139619-16590bf6-1893-4292-ae9e-f0eb08d88c64.png)

**Comment by testpaul999** [comment]
Das Problem mit der Stabilität des CAN- Bus habe ich mit einem neuen Adapter gelöst.
![image](https://user-images.githubusercontent.com/35597907/219140313-54bdda4c-f1a6-40af-8945-1f84bfad660a.jpg)
Damit ist die Verbindung stabil.
Das war der alte Adapter 
![image](https://user-images.githubusercontent.com/35597907/219140614-65ef5557-0f51-4735-947c-13cc1d3dfe41.jpg)

**Reply by shining-man** [reply]
Das ist der Grund warum es die BSC Hardware gibt. Wenn ich vom xxxx€ mir einen Speicher baue, dann kommt es auf die xx€ auch nicht mehr darauf an. Ich habe jedefalls eine Hardware der ich trauen kann.

**Reply by testpaul999** [reply]
Das ist richtig! Habe auch im Forum einen geordert.
Ich brauche halt nur für einen Akku etwas und ich bastele (löte) gerne selber. 😉
Grüße

**Comment by shining-man** [comment]
[firmware.zip](https://github.com/shining-man/bsc_fw/files/10747374/firmware.zip)
Teste einmal ob die Werte jetzt i.O. sind.

**Reply by testpaul999** [reply]
👍🏻😀 Strom stimmt.
Danke.

**Comment by bennyb21** [comment]
Ich habe noch mal eine Grundsätzliche Frage zur JK-BMS Anbindung. Das JK hat ja direkt eine TTL-Schnittstelle. Also 3,3V. Ist das mit dem BSC berücksichtigt bzw. kann man das entsprechend einstellen? Hatte in der Doku dazu nichts gefudnen oder war zu Blind. Wenn man da mit dem RS485 direkt drauf geht sind es ja meine ich 5V. Hatte das mal versehentlich bei einem USB-TTL Adapter auf 5V eingestellt und da lieferte das BMS zum Teil grütze zurück (Temperaturwerte waren komplett falsch). Oder bin ich gerade Gedanklich ganz falsch abgebogen?!

**Reply by shining-man** [reply]
Kann es sein, dass du TTL und RS485 durcheinander bringst? RS485 hat nichts mit TTL zu tun. 
Ich habe kein JK-BMS, aber so viel ich weiß hat das JK-BMS zwei Schnittstellen:
- Eine RS485 Schnittstelle und 
- eine RS232 Schnitstelle mit 3,3V TTL Pegel.

Wenn du die RS485 Schnittstelle benutzt brauchst du die keine Gedanken machen ob A/B 3 V oder 5 V Highpegel haben. Die Transceiver können das. Der von mir verwendete SP3485 ist ein 3V3 Typ. Wenn du die RS485 Schnittstelle des JK nehmen willst, dann muss beim BSC der U16 bestückt sein (Ich verwende für das Beispiel die Serial 1, was im Stromlaufplan U2RX/TX ist).

Wenn du die RS232 Schnittstelle des JK-BMS verwenden möchtest, dann darf beim BSC der U16 (Als Beispiel wieder die Serial 1) nicht bestückt sein. Statt dessen musst du die Lötjumber JP18 und JP19 brücken.

Es ist zu beachten, dass an der Serial 0 ein Problem vorliegt. Hier muss ein Pullup (1k) am GPIO17 (Serial 0 TX) eingefügt werden, da der ESP sonst nur einen Pegel von ca. 2V schafft. Dies werde ich aber noch veröffentlichen, sobald ich es genauer untersucht habe. 
Wenn nicht alle drei serielle benötigt werden, würde ich Serial 1 oder 2 verwenden. Wenn alle 3 benötigt wereden, dann muss ein Pullup rein.

Schaltplanauszug von der Serial 1:
![grafik](https://user-images.githubusercontent.com/109786660/224792311-fd2540be-310b-4044-aecf-2ac990a897e2.png)

**Comment by bennyb21** [comment]
Danke für die Antwort. 3,3V wäre ja richtig.
Ich habe jetzt auch noch mal ein wenig geschaut. Man muss beim JK wohl unterscheiden. Direkt am BMS ist wohl eine UART-TTL Schnittstelle. Die kann man (was in anderen Projekten auch gemacht wird) direkt an einen ESP hängen. Es wird zu den BMS auch gerne ein kleiner Dongle verkauft für RS485. Der soll die galvanische Trennung bringen und als ein Pegelwandler funktionieren. Mehr Schnittstellen hat das BMS nicht. 

Ich selbst benötige tatsächlich wohl alle 3 seriellen Schnittstellen (2x BMS 1x WR Deye). Das heißt, ich müsste dann mit dem Pullup arbeiten.

Brauchst noch wen zum testen? :) 

Ich werde wohl am Wochenende den Deye an die Wand bringen und dann ist das Setup fertig. Die beiden Akkus sind bereits fertig und laufen derzeit noch an einem Victron MP. 

Wenn ich jetzt alles richtig verstanden habe, könnte ich es in einem Testsetup mit dem bekannten Problem der fehlenden galvanischen Trennung auch alles fliegend aufbauen, sofern ich mir noch einen CAN-Adapter besorge. Die beiden BMS könnte ich direkt verbinden mit dem ESP, da 3,3V. Den Deye bringe ich mit dem CAN-Adapter an den ESP. Software flashen und es könnte funktionieren.

Gibt es eine Info, welcher CAN-Adapter getestet funktioniert und einigermaßen verfügbar ist?

**Comment by testpaul999** [comment]
@bennyb21 
Ich habe gute Erfahrungen mit diesem can Adapter 
https://www.amazon.de/gp/product/B09GF968MH/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1
Habe das auch fliegend aufgebaut, zum Testen.
![image](https://user-images.githubusercontent.com/35597907/224926829-823f9d6e-949d-47b5-8e75-c1759dfec9ad.jpg)

Dieser esp32 
https://www.amazon.de/gp/product/B0BFDKZT9Q/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1
damit ist die Bluetooth Verbindung stabiel.
Grüße

**Reply by bennyb21** [reply]
Sehr gut! Danke! Von den ESP´s habe ich noch einige rumfliegen. Beim CAN-Adapter war ich noch unschlüssig... :) Ich gehe mal davon aus, auf Deinem Raspi läuft dann VenusOS für Victron?!

**Reply by testpaul999** [reply]
Ja sicher v3.0~23 🙂

**Reply by bennyb21** [reply]
Du nutzt derzeit aber nur 2 serielle Schnittstellen, oder? Dann betrifft Dich die Geschichte mit dem Pullup auf GPIO17 nicht?!

**Reply by testpaul999** [reply]
Hallo, trifft mich wirklich nicht, weil ich sowieso nur ein BMS dran habe.

**Reply by bennyb21** [reply]
Okay. Und dann hoffentlich die letzte Frage: Bislang habe ich es so verstanden, dass eine der seriellen Schnittstellen dann im Menü für den WR als CAN definiert wird. Jetzt habe ich oben in dieser Diskussion gelesen, dass WR-CAN andere GPIO´s verwenden. Also muss der CAN-Adapter an die GPIO (4 und 5 meine ich)?

**Reply by testpaul999** [reply]
Ja siehe hier https://github.com/shining-man/bsc_fw/discussions/24#discussioncomment-4909363
Und hier
https://github.com/shining-man/bsc_fw/discussions/24#discussioncomment-4950879

**Reply by bennyb21** [reply]
Sehr gut. Dann habe ich ja doch nicht das Problem mit dem Pullup. 

Wenn die Infos jetzt alle Stimmen:

//Serial 0
#define SERIAL1_PIN_RX 16
#define SERIAL1_PIN_TX 17
#define SERIAL1_PIN_TX_EN 18

//Serial 1
#define SERIAL2_PIN_RX 23
#define SERIAL2_PIN_TX 25
#define SERIAL2_PIN_TX_EN 2

//Serial 2
#define SERIAL3_PIN_RX 35
#define SERIAL3_PIN_TX 33
#define SERIAL3_PIN_TX_EN 32

CAN:
TX: GPIO4
RX: GPIO5

Dann nutze ich ja 4+5 für CAN und kann normal Serial 1 und 2 für die beiden BMS nutzen ohne den Pullup auf GPIO17.

Der Adapter wird die Woche da sein. Ich hoffe mal am WE geht alles gut mit dem WR tausch und dann teste ich das mal aus. :)

Danke schon mal für die Infos!!!

**Reply by shining-man** [reply]
Die Infos stimmen alle.
Wenn du nur zwei serielle brauchst, dann nutze einfach Serial 1 und 2.

**Reply by bennyb21** [reply]
So... Ich habe es am Wochenende geschafft den WR in Betrieb zu nehmen und habe mich auch an den Testaufbau zum BSC gemacht. 
Dazu habe ich gleich das Release 3.1 genommen. Die Serial-Verbindungen zu beiden JK-BMS steht. Die Daten landen im MQTT, alles schick!!! Allerdings Verzweifel ich ein wenig am CAN... Ich habe da noch keine funktionierende Verbindung hinbekommen. 

GPIO 54 und 5 genommen. TX/RX mal gekreuzt oder nicht gekreuzt. CAN H/L mal gedreht... Nichts. Im Log habe ich immer folgenden Fehler oder Hinweis: 

`E (7707281) CAN: TX error: REG_ECC=0, REG_SR=1`

Ist das ein Fehler noch auf Seiten des CAN-Adapter? Oder ist das eher ein Hinweis auf eine fehlende Verbindung zum WR?

**Reply by shining-man** [reply]
Was für einen CAN-Adapter hast du denn?
CAN H/L gehören nicht gekreuzt.

**Reply by bennyb21** [reply]
Ich habe den von @testpaul999 getesteten ebenfalls bestellt. Meine den von Waveshare. Jupp: Waveshare SN65HVD230

Ja, ist mir bewusst das die nicht gekreuzt werden sollen, hatte es nur getestet, falls die Pinbelegung im RJ45 zum WR nicht gestimmt hätte... ;)

**Reply by shining-man** [reply]
Sollte mit einer 1:1 Vernindung eigentlich funktionieren. Auch RX/TX gehören nicht gekreuzt.
Hat das Waveshare SN65HVD230 oder der WR einen Abschlusswiderstand? Wenn nicht würde ich am SN65HVD230 jedenfalls einmal einen 120Ohm zwischen CAN H und CAN L machen.
Ändert sich die Fehlermeldung wenn du den Wechselrichter nicht an den SN65HVD230 anschließt?
An den richtigen Pins am DEYE bist du?

**Reply by bennyb21** [reply]
Nein, die Fehlermeldung ist immer identisch.

Pins am Deye sollten korrekt sein. Habe ich in zwei unterschiedlichen Dokumentationen so gefunden. Mehrfach kontrolliert...

Ob der Waveshare oder der Deye einen Abschlusswiderstand haben, das kann ich nicht sagen, das mit dem Widerstand kann ich noch mal testen!

**Reply by bennyb21** [reply]
So... Mit Widerstand auch getestet. Leider kein Erfolg. Den Waveshare Adapter gabs ja im Doppelpack. Ist bei beiden identisch...

Im Akkudoktor-Forum hat es ja bereits jemand zum laufen bekommen. Habe ihn noch mal angeschrieben ob er noch einen Tipp hat. Aber soweit ich das gelesen habe, hat er auch direkt die BSC-Platine genutzt. Aber das dauert noch ein paar Tage bis ich bei der nächsten Sammelbestellung zum Zug komme... ;)

**Reply by shining-man** [reply]
Prinzipiell sollte es aber auch so gehen. Wenn es auch immer wieder zu Schwierigkeiten und Problemen führt.
Welchen ESP32 verwendest du denn?

**Comment by bennyb21** [comment]
Ich nutze die ESP32 Dev Kit C V4 NodeMCU. Ich habe eh noch welche nachbestellt, da ich den letzten aus meinem Fundus genutzt habe. Die kommen auch heute an, werde es einfach noch mal mit einem anderen testen...

**Reply by bennyb21** [reply]
Noch mal als Rückmeldung: Ich habe den Testaufbau mit einem neuen ESP, gleicher Hersteller wie vorher, noch einmal sauber neu aufgebaut. Firmware drauf. Alles bestens. Lief sofort einwandfrei. 
Entweder war der daher von vornherein defekt, oder ich habe ihn ein wenig zu sehr geärgert mit der ganzen Testerei. 

Danke für die Hilfe!!! Jetzt werde ich mich mal an die Einstellungen und Alarme machen. Was mir dabei einfällt... Es sind ja am Ende bei einer aufwändigen Config schon einige Einstellungen. Hast Du schon mal über eine Backup-Funktion nachgedacht? Macht es Sinn dafür mal ein Issue aufzumachen oder ist das zu weit out of Scope?

**Reply by shining-man** [reply]
Natürlich kannst du dafür ein Issue aufmachen. Der Download des Configfiles geht mittels "/param".

**Reply by testpaul999** [reply]
😱 das lässt ja gleich eine Diskussion aufkommen, Passworte im Klartext.
Lässt sich daran was anpassen?

Danke

**Reply by shining-man** [reply]
Das ist der Grund warum die Funktion bisher nicht public ist und es so im einem neuen Release auch nicht geben wird. 
Das sollte so nicht ins Release sondern war nur für Testzwecke.
Wenn jemals ein Parameter-export kommt, dann ohne die Passwort Parameter oder verschlüsselt, das weiß ich noch nicht.
Es kann aber auch gerne ein Pull-Request, der das Problem behebt, kommen.

**Reply by shining-man** [reply]
In den letzten Testständen ist die Funktion schon rausgenommen.
